src/GraphQL.ApiTests/GraphQL.MicrosoftDI.approved.txt

```diff
@@ -51,11 +51,7 @@ namespace GraphQL.MicrosoftDI
         public GraphQL.Builders.FieldBuilder<TSourceType, TReturnType> ResolveAsync(System.Func<GraphQL.IResolveFieldContext<TSourceType>, T1, T2, T3, T4, T5, System.Threading.Tasks.Task<TReturnType>> resolver) { }
         public GraphQL.MicrosoftDI.ResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithScope() { }
     }
-    public class ScopedAsyncFieldResolver<TReturnType> : GraphQL.Resolvers.AsyncFieldResolver<TReturnType>
-    {
-        public ScopedAsyncFieldResolver(System.Func<GraphQL.IResolveFieldContext, System.Threading.Tasks.Task<TReturnType>> resolver) { }
-    }
-    public class ScopedAsyncFieldResolver<TSourceType, TReturnType> : GraphQL.Resolvers.AsyncFieldResolver<TReturnType>
+    public class ScopedAsyncFieldResolver<TSourceType, TReturnType> : GraphQL.Resolvers.AsyncFieldResolver<TSourceType, TReturnType>
```

Comment:
I designed this to be flexible and reduce allocations.  With this design there is always an extra allocation required.  I will need to review my thought process and use cases.  Same thought applies to `ScopedFieldResolver`.

Reply:
Does the same apply to connection builder? I'm ok either way but both builders should act the same IMO - with or without TSourceType.

Reply:
Yes if it was public, although I had made it internal for reasons previously stated (and then eliminated redundant code). We can make it public if we expose bidirectional and similar as `get;` properties on the builder. This also eliminates an allocation.

Reply:
By the way, as for builders I'm not so concerned about allocations.

Reply:
And regarding these 3 properties in ConnectionBuilder. In fact, these are calculated properties - `_isUnidirectional` and `_isBidirectional`. The real object that is configured by builder is `FieldType`. `_pageSize` is not calculated but could be stored in `FieldType.Metadata` so `ConnectionBuilder` will store only reference to `FieldType` as `FieldBuilder` does. `IsUnidirectional` and `IsBidirectional` can be implemented in such a way that they set the arguments only if necessary.

Reply:
The allocations I was referring to occur during execution. I am not counting allocations during build time.